SOLID_No2.php

```diff
@@ -0,0 +1,49 @@
+<!DOCTYPE html>
+<html>
+<body>
+  <?php
+  interface WorkAbleInterface{
+      public  function work();
+  }
+
+  interface SleepAbleInterface{
+      public  function  sleep();
+  }
+
+  interface Electricity{
+    public function hasPower();
+  }
+
+  class Workers implements WorkAbleInterface{
+
+    public  function work(){
+    var_dump('works');
+  }
+  }
+
+  class HumanWorker extends Workers implements SleepAbleInterface{
+
+      public  function  sleep(){
+          var_dump('sleep');
+      }
+
+  }
+
+  class RobotWorker extends Workers implements Electricity{
```

Comment:
Please checking in here. RobotWorker violated Liskov Substitution Principle.
Because RobotWorker->work function need to check function RobotWorker->hasPower before calling parent::work();
Note: Important point in LSP, calling parent should be before doing anything.
Please removing extends Workers .

Reply:
Fixed. Please check again.